### PR TITLE
add INSA by NASA

### PR DESCRIPTION
**Pull request by jfilter**

### CODE REVIEW

File: p/insa/example.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	// client.location('008010226') // Magdeburg-Neustadt
+	// client
+	// 	.nearby(
+	// 		{
+	// 			type: "location",
+	// 			latitude: 52.148842,
+	// 			longitude: 11.641705
+	// 		},
+	// 		{ distance: 200 }
+	// 	)
```
Comment (derhuerst) [inline p/insa/example.js]: Please indent this block like in the other example files.

File: test/insa.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
+"use strict";
+
+const tapePromise = require("tape-promise").default;
+const tape = require("tape");
+const isRoughlyEqual = require("is-roughly-equal");
+const validateFptf = require("validate-fptf");
+
+const co = require("./co");
```
Comment (derhuerst) [inline test/insa.js]: Use single quotes wherever possible. See https://standardjs.com/rules.html

File: test/insa.js [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+	assertValidStopover,
+	hour,
+	createWhen,
+	assertValidWhen
+} = require("./util.js");
+
+const when = createWhen("Europe/Berlin", "de-DE");
+
+const assertValidStationProducts = (t, p) => {
+	return null; // todo
```
Comment (derhuerst) [inline test/insa.js]: This needs to be fixed. 😉

File: test/insa.js [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+	for (let k of Object.keys(products)) {
+		t.ok("boolean", typeof products[k], "mode " + k + " must be a boolean");
+	}
+};
+
+const test = tapePromise(tape);
+const client = createClient(insaProfile);
+
+test(
+	"Magdeburg Hbf to Magdeburg-Buckau",
```
Comment (derhuerst) [inline test/insa.js]: Please indent this tests like e.g. `test/oebb.js`.

**Comment by derhuerst** [comment]
Thanks a lot! ❤️

Things still to do:

- [ ] check if the endpoint supports `journeyLeg`
- [ ] check if the endpoint supports `radar`
- [ ] remove some todos in `test/insa.js`

**Comment by derhuerst** [comment]
@jfilter friendly ping (:

**Comment by derhuerst** [comment]
There are only the `via` tests failing now, which are flaky with the other profiles as well. They will be replaced with something more reliable as of #26.

**Comment by jfilter** [comment]
Thanks @derhuerst for all the work.

**Pull request merged**